src/renderer/store/modules/settings.js

```diff
@@ -234,11 +234,31 @@ const stateWithSideEffects = {
       let targetLocale = value
       if (value === 'system') {
         const systemLocale = await dispatch('getSystemLocale')
-
-        targetLocale = Object.keys(i18n.messages).find((locale) => {
-          const localeName = locale.replace('-', '_')
-          return localeName.includes(systemLocale.replace('-', '_'))
+        const systemLocaleName = systemLocale.replace('-', '_')
```

Comment:
Maybe good to put example value(s) in code comments (whole section not just this line)
I can't understand this code section easily